File: shift-only/shift-only.cpp
#include <bits/stdc++.h>

// shift-only.cpp
// author: Seong Yong-ju ( @sei40kr )
// Copyright (c) 2017 Seong Yong-ju

#define REP(i, n) for (int i = 0; i < (n); ++i)

using namespace std;

int main() {
  cin.tie(nullptr);
  ios::sync_with_stdio(false);

  int n;
  cin >> n;

  int ans = INT_MAX;
  REP(i, n) {
    int x;
    cin >> x;

    int cnt = 0;

    while (x % 2 == 0) {
      x /= 2;
      ++cnt;
    }

    ans = min(ans, cnt);
  }

  cout << ans << endl;
}

File: shift-only/shift-only.hs
-- shift-only.hs --- Shift Only

-- author: Seong Yong-ju <sei40kr@gmail.com>

f :: Int -> Int
f a =
  if mod a 2 == 0
    then 1 + f (quot a 2)
    else 0

solve :: [Int] -> Int
solve = minimum . map f

main :: IO ()
main = do
  n <- (read :: String -> Int) <$> getLine
  as <- map (read :: String -> Int) . take n . words <$> getLine
  print $ solve as
